File: TravellerCreatorModels/StarFrontiers/StarFrontiersStarSystem.cs
﻿using TravellerCreatorModels.Enums;
using TravellerCreatorModels.Interfaces;

namespace TravellerCreatorModels.StarFrontiers;

public class StarFrontiersStarSystem : IStarSystem
{
    public List<IStar> Stars { get; set; } = new();
    public StarSystemType Type { get; set; }
    public bool GasGiant { get; set; }
    public IPlanet? Planet { get; set; }
    public List<IPlanet> Planets { get; set; } = new();
    
    public void GetGasGiant()
    {
        throw new NotImplementedException();
    }
}

File: TravellerCreator/ClientApp/src/app/sector-creator/models/star-frontiers/star-frontiers-star-system.ts
import {IStarSystem} from "../interfaces/star-system";
import {StarSystemType} from "../enums/star-system-type";
import {StarFrontiersStar} from "./star-frontiers-star";
import {StarFrontiersPlanet} from "./star-frontiers-planet";

export class StarFrontiersStarSystem implements IStarSystem {
  stars: StarFrontiersStar[];
  type: StarSystemType;
  gasGiant: boolean;
  planet: StarFrontiersPlanet;
  planets: StarFrontiersPlanet[];
}
